### PR TITLE
feat:0117 update filter.py and model.py

### PR DESCRIPTION
**Pull request by SeJinKIM8845**

### CODE REVIEW

File: consumer/filter.py [hunk 1]
```diff
@@ -1,542 +1,316 @@
...
-       # 시뮬레이션 ID 무작위화
-       if 'simulation_id' in data.columns:
-           unique_ids = data['simulation_id'].unique()
-           id_mapping = {old_id: f"sim_{i:03d}" for i, old_id in enumerate(unique_ids)}
-           data['simulation_id'] = data['simulation_id'].map(id_mapping)
+   def _shuffle_data(self, sequences: np.ndarray) -> np.ndarray:
+       """Shuffle sequences randomly"""
+       indices = np.arange(sequences.shape[0])
+       np.random.shuffle(indices)
+       shuffled_sequences = sequences[indices]
```
Comment (minz95) [inline consumer/filter.py]: shuffle 은 simulation 끼리의 순서만 바꿀 수 있어야 합니다
현재 `_shuffle_data` 에서는 simulation 내에서 time series 의 순서가 섞여서 의미 없는 데이터가 될 것 같습니다

File: consumer/filter.py [hunk 2]
```diff
@@ -1,542 +1,316 @@
...
+    """Normalize data and store scaling parameters"""
+    normalized_data = {}
+    scaler_params = {}
+    
+    # feature_range를 tuple로 변환
+    feature_range = tuple(self.config['preprocessing']['normalization']['feature_range'])
+    
+    for sensor_type, df in data.items():
+        features = self.config['sensors']['primary'].get(sensor_type, {}).get('features', []) or \
+                  self.config['sensors']['secondary'].get(sensor_type, {}).get('features', [])
```
Comment (minz95) [inline consumer/filter.py]: sequence 별 normalize 가 아니라 simulation 별 normalize 가 되는게 좋을 것 같습니다

File: consumer/model.py [hunk 3]
```diff
@@ -1,208 +1,280 @@
...
+            self.model.parameters(),
+            lr=self.config['training']['learning_rate']
+        )
+
+    def prepare_data(self) -> Tuple[DataLoader, DataLoader]:
+        """Prepare data for training and testing"""
+        data_path = Path(self.config['data']['processed_data_path'])
+        sequences = np.load(data_path / 'batch_sequences.npy')
+        
+        labels = self._create_anomaly_labels(sequences)
```
Comment (minz95) [inline consumer/model.py]: 여기 labels 를 position_x, position_y, position_z 로 수정해야 합니다

File: consumer/model.py [hunk 4]
```diff
@@ -1,208 +1,280 @@
...
+            nn.Tanh(),
+            nn.Linear(hidden_size, 1)
+        )
         
-        # LSTM 레이어
+        self.fc = nn.Sequential(
+            nn.Linear(hidden_size * 2, hidden_size),
+            nn.ReLU(),
+            nn.Dropout(dropout),
+            nn.Linear(hidden_size, 1)
```
Comment (minz95) [inline consumer/model.py]: 여기서 output 이 3개 이므로 3 으로 바꾸셔야 합니다

File: consumer/model.py [hunk 5]
```diff
@@ -1,208 +1,280 @@
...
-        """
-        모델 학습
-        Args:
-            train_loader: 학습 데이터 로더
-        Returns:
-            학습 손실 리스트
-        """
-        self.model.train()
-        losses = []
+    def _create_anomaly_labels(self, sequences: np.ndarray) -> np.ndarray:
```
Comment (minz95) [inline consumer/model.py]: consumer 에서 model 을 호출하면 (x, y, z) 값을 얻습니다. 
그 후에 이 함수를 사용해서 anomaly 여부를 판별해서 return 합니다.
ex) check_anomaly.py -> 저장된 모델 파일을 로드해서 갖고 있고 (모델 object 를 들고 있다) 해당 모델을 사용해서 새로 들어온 데이터를 사용해 경로를 예측하고 anomaly 여부를 판별해서 리턴해줍니다.
